.github/workflows/docs.yml

```diff
@@ -0,0 +1,53 @@
+name: Build and deploy docs
+
+on:
+  push:
+    branches:
+      - pr1-gh-docs # tmp testing # todo: remove
+      - main
+    tags:
+      - '*.*.*'
+  # todo: remove below once test succeeds
+  pull_request:
+    branches:
+    - main
+    tags:
+      - '*.*.*'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    name: Xonsh docs to gh-pages
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.9'
+          cache: 'pip'
+          cache-dependency-path: 'requirements/docs.txt'
+      - run: |
+          pip install -r requirements/docs.txt
+          pip install . --no-deps
+      - name: Build Docs
+        run: cd docs && make html
+      - name: Deploy to dev if not tagged
+        if: "!startsWith(github.ref, 'refs/tags/v')"
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+#          external_repository: xonsh/xonsh-docs
+          publish_dir: ./docs/_build/html
+          destination_dir: dev # latest as in main branch
+      - name: Deploy to stable if tagged
+        if: startsWith(github.ref, 'refs/tags/v')
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          #          external_repository: xonsh/xonsh-docs
+          publish_dir: ./docs/_build/html
```

Comment:
@gforsyth  good, I just adjusted the it to match the app. It is failing because of the missing secrets. Could you explain what is app-id and private-key parameters?

Reply:
Those are the credentials to use the xonsh-org github app to generate a token for usage in a separate step.

These only work if the PR is not on a fork

Reply:
OK. I have removed the tmp code changes. Lets see if it deploys to dev folder correctly.